Backend/WebApi-PointMap/Controllers/UserManagementController.cs

```diff
@@ -198,140 +169,111 @@ public IHttpActionResult DeleteUser([FromBody, Required] LoginDTO requestPayload
         public IHttpActionResult UpdateUser([FromBody] UpdateUserRequestDTO payload)
-        {
-            var token = GetHeader(Request, "Token");
-            if (token.Length < 1)
-            {
-                return Content(HttpStatusCode.Unauthorized, "No token provided.");
-            }
-            if (!ModelState.IsValid || payload == null)
-            {
-                return Content((HttpStatusCode)412, ModelState);
-            }
-            Guid UserId;
-            try
-            {
-                // check if valid SSO ID format
-                UserId = Guid.Parse(payload.id);
-            }
-            catch (Exception)
-            {
-                return Content((HttpStatusCode)400, "Invalid User ID");
-            }
-            using (var _db = new DatabaseContext())
-            {
-                SessionService _sessionService = new SessionService();
-                var session = _sessionService.ValidateSession(_db, token);
-                if (session == null)
-                {
-                    return Content(HttpStatusCode.NotFound, "Session is no longer available.");
-                }
+        {
+            try
+            {
+                var token = GetAndCheckToken(Request, "Token");
+                if (!ModelState.IsValid || payload == null)
+                {
+                    return Content((HttpStatusCode)412, ModelState);
+                }
+                _db = new DatabaseContext();
+
+                var UserId = ParseAndCheckId(payload.Id);
+
+                var session = ValidateAndUpdateSession(_db, token);
+
                 UserManagementManager _userManager = new UserManagementManager(_db);
                 var byUser = _userManager.GetUser(session.UserId);
                 if (byUser.IsAdministrator)
                 {
                     var user = _userManager.GetUser(UserId);
                     if (user == null)
                     {
-                        return Content(HttpStatusCode.NotFound, "User does not exist.");
+                        throw new NullReferenceException("User does not exist.");
                     }
-                    user.City = payload.city;
-                    user.State = payload.state;
-                    user.Country = payload.country;
-                    user.Disabled = payload.disabled;
-                    user.IsAdministrator = payload.isAdmin;
-                    try
-                    {
-                        var ManagerId = Guid.Parse(payload.manager);
+                    user.City = payload.City;
+                    user.State = payload.State;
+                    user.Country = payload.Country;
+                    user.Disabled = payload.Disabled;
+                    user.IsAdministrator = payload.IsAdmin;
+                    user.ManagerId = null;
+                    if (payload.Manager != null)
+                    {
+                        var ManagerId = Guid.Parse(payload.Manager);

```

Comment:
No need to check the parse, as this is already in the database and is therefore correct.